hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/DownloadAndImportReplicator.java

```diff
@@ -125,9 +126,10 @@ public void replicate(ReplicationTask task) {
         importContainer(containerID, path);
         LOG.info("Container {} is replicated successfully", containerID);
         task.setStatus(Status.DONE);
-      } catch (Exception e) {
+      } catch (InterruptedException | IOException | ExecutionException e) {
         LOG.error("Container {} replication was unsuccessful.", containerID, e);
         task.setStatus(Status.FAILED);
+        Thread.currentThread().interrupt();
```

Comment:
curious... why do we set thread's interrupt flag when there's an IOException?

Reply:
Thanks to @jojochuang 's review, it is a better choice to handle InterruptedException and IOException separately rather than together.Let me do it.